src/checkers/inference/InferrableAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,22 @@
+package checkers.inference;
+
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+import org.checkerframework.framework.type.treeannotator.TreeAnnotator;
+
+/**
+ * Interface for all annotatedTypeFactory that wish to be used with
+ * Checker-Framework-Inference
+ */
+
+public interface InferrableAnnotatedTypeFactory {
+
+    /**
+     * return the TreeAnnotator from real type annotatedTypeFactory
+     * 
+     */
+    TreeAnnotator getRealInferenceTreeAnnotator(
```

Comment:
Document that the returned TreeAnnotator should contain a subtype of InferenceTreeAnnotator and that a ListTreeAnnotator needs to be used for implicits.
Refer `@see` to the default implementation if this method is not implemented.